### PR TITLE
App 1175: Cryptography service

### PR DESCRIPTION
**Pull request by campidelli**

### CODE REVIEW

File: integration-commons/src/main/java/org/symphonyoss/integration/service/KeyManagerService.java [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  /**
+   * Return the Key Manager data for the given user.
+   * @param userId The bot user ID.
+   * @return Key manager user data.
+   */
+  UserKeyManagerData getBotUserAccountKeyByUser(String userId);
+
+  /**
+   * Return the Key Manager data for the given configuration ID.
+   * @param userId The configuration ID.
```
Comment (robson-sanchez) [inline integration-commons/src/main/java/org/symphonyoss/integration/service/KeyManagerService.java]: Change this parameter to 'configurationId'.

File: integration-commons/src/main/java/org/symphonyoss/integration/service/KeyManagerService.java [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+ * Created by campidelli on 9/11/17.
+ */
+public interface KeyManagerService {
+
+  /**
+   * Return the Key Manager data for the given user.
+   * @param userId The bot user ID.
+   * @return Key manager user data.
+   */
+  UserKeyManagerData getBotUserAccountKeyByUser(String userId);
```
Comment (robson-sanchez) [inline integration-commons/src/main/java/org/symphonyoss/integration/service/KeyManagerService.java]: I think this method isn't required to be part of the interface.

File: integration-commons/src/main/java/org/symphonyoss/integration/service/KeyManagerService.java [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+  /**
+   * Return the Key Manager data for the given configuration ID.
+   * @param userId The configuration ID.
+   * @return Key manager user data.
+   */
+  UserKeyManagerData getBotUserAccountKeyByConfiguration(String userId);
```
Comment (robson-sanchez) [inline integration-commons/src/main/java/org/symphonyoss/integration/service/KeyManagerService.java]: We should move this method to the UserService interface that defines methods to get user information.

I think we could rename the method to 'getBotUserAccountKeyByConfigurationId'.

**Review by robson-sanchez — CHANGES_REQUESTED** [review]

**Comment by robson-sanchez** [comment]
@campidelli Travis CI build failed caused by test code coverage less than 90%

File: integration-commons/src/main/java/org/symphonyoss/integration/service/KeyManagerService.java (hunk 2 above)
Comment (campidelli) [reply integration-commons/src/main/java/org/symphonyoss/integration/service/KeyManagerService.java]: Done.

File: integration-commons/src/main/java/org/symphonyoss/integration/service/KeyManagerService.java (hunk 1 above)
Comment (campidelli) [reply integration-commons/src/main/java/org/symphonyoss/integration/service/KeyManagerService.java]: Done.

File: integration-commons/src/main/java/org/symphonyoss/integration/service/KeyManagerService.java (hunk 3 above)
Comment (campidelli) [reply integration-commons/src/main/java/org/symphonyoss/integration/service/KeyManagerService.java]: Done.

**Review by robson-sanchez — APPROVED** [review]

**Pull request merged**